builtin/plugins/coin/coin.go

```diff
@@ -209,10 +254,223 @@ func mint(ctx contract.Context, to loom.Address, amount *loom.BigUInt) error {
 	if err := emitTransferEvent(ctx, mintAddress, to, amount); err != nil {
 		return err
 	}
-
 	return ctx.Set(economyKey, econ)
 }
 
+//Mint : to be called by CoinPolicyManager Responsible to mint coins as per various parameter defined
+func Mint(ctx contract.Context) error {
+	div := loom.NewBigUIntFromInt(10)
+	div.Exp(div, loom.NewBigUIntFromInt(18), nil)
+	var policy Policy
+	var amount *common.BigUInt
+	err := ctx.Get(policyKey, &policy)
+	if err != nil {
+		return errUtil.Wrap(err, "Failed to Get PolicyInfo")
+	}
+	changeRatioNumerator := loom.NewBigUIntFromInt(int64(policy.ChangeRatioNumerator))
+	changeRatioDenominator := loom.NewBigUIntFromInt(int64(policy.ChangeRatioDenominator))
+	blockHeight := loom.NewBigUIntFromInt(ctx.Block().Height)
+	baseAmount := loom.NewBigUIntFromInt(int64(policy.TotalSupply))
+	baseAmount.Mul(baseAmount, div)
+	blocksGeneratedPerYear := loom.NewBigUIntFromInt(int64(policy.BlocksGeneratedPerYear))
+	basePercentage := loom.NewBigUIntFromInt(int64(policy.BasePercentage))
+	operator := policy.Operator
+	//Checks if minting amount is computed before or it is being computed for first time
+	var checkMintAmount = &types.BigUInt{
+		Value: *loom.NewBigUIntFromInt(0),
+	}
+	err = ctx.Get(mintingAmountKey, checkMintAmount)
+	if err != nil && err != contract.ErrNotFound {
+		return err
+	}
+	if err == contract.ErrNotFound {
+	//Sets Minting Height - Height at which minting started
+		err := ctx.Set(mintingHeightKey, &types.BigUInt{
+			Value: *loom.NewBigUIntFromInt(ctx.Block().Height),
+		})
+		if err != nil {
+			return err
+		}
+	}
+	var mintingHeight = &types.BigUInt{
+		Value: *loom.NewBigUIntFromInt(0),
+	}
+	err1 := ctx.Get(mintingHeightKey, mintingHeight)
```

Comment:
Why `err1`?

Reply:
I need to track previous error in later part of the code, so need to use this new error variable here.
First err is for mintingAmount, Second err is for mintingHeight